src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -111,6 +112,31 @@ public static LocalDateTime parseDate(String date) throws ParseException {
         return time;
     }
 
+    /**
+     * Parses a {@code String date} into a {@code LocalDate}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code date} is invalid or not in the expected format.
+     */
+    public static LocalDate parseDayDate(String date) throws ParseException {
+        requireNonNull(date);
+        String trimmedDate = date.trim();
+        LocalDate parsedDate;
+        LocalDate currentDate = LocalDate.now();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        try {
+            parsedDate = LocalDate.parse(trimmedDate, formatter);
+        } catch (DateTimeParseException e) {
+            throw new ParseException("Invalid date format, please use yyyy-MM-dd.");
+        }
+
+        if (currentDate.isAfter(parsedDate)) {
+            throw new ParseException("Invalid date entered. The date can't be in the past!");
+        }
```

Comment:
May not be necessary, but it's not wrong

Reply:
Do I need to change it ? Or just keep it